### ISSUE TITLE
False negatives in Layout/ClassStructure with inline access modifiers

### ISSUE THREAD

**Issue by dduugg** [rubocop-hq/rubocop#9415]
The `Layout/ClassStructure` cop fails to surface private/public method ordering violations when inline access modifiers are used. Given the file `ab.rb` with:

```
class A
  private def foo
  end

  public def bar
  end
end

class B
  private
  def foo
  end

  public
  def bar
  end
end
```
--------

## Expected behavior

`Layout/ClassStructure` should be agnostic to visibility modifier and flag both definitions of `bar` (`A#bar` and `B#bar`).

## Actual behavior
Only `B#bar` is flagged:
```
rubocop ab.rb --only Layout/ClassStructure
Inspecting 1 file
C

Offenses:

ab.rb:15:3: C: [Correctable] Layout/ClassStructure: public_methods is supposed to appear before private_methods.
  def bar ...
  ^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## Steps to reproduce the problem

☝️ 

## RuboCop version

```
1.8.1 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-darwin20)
  - rubocop-performance 1.9.2
  - rubocop-rake 0.5.1
  - rubocop-rspec 2.1.0
  - rubocop-sorbet 0.5.1
```

**Comment by AndreiEres** [comment]
@dduugg, @koic 
Could you have a look to that fix https://github.com/rubocop-hq/rubocop/pull/9419?

**Comment by dduugg** [comment]
Wow, fast work @AndreiEres! I left one comment, regarding implicit public methods.

**Comment by AndreiEres** [comment]
@dduugg I suppose I fixed it
